docker/pravega/scripts/init_kubernetes.sh

```diff
@@ -47,17 +47,20 @@ init_kubernetes() {
         local podname=${POD_NAME}
         export PUBLISHED_ADDRESS=""
         export PUBLISHED_PORT=""
-
+	
+	export PUBLISHED_ADDRESS=$( k8 "${ns}" "services" "${podname}" ".metadata.annotations[\"external-dns.alpha.kubernetes.io/hostname\"]" )
         service_type=$( k8 "${ns}" "services" "${podname}" ".spec.type" )
         if [ "${service_type}" == "LoadBalancer" ]; then
             while [ -z ${PUBLISHED_ADDRESS} ] || [ -z ${PUBLISHED_PORT} ]
             do
                 echo "Trying to obtain LoadBalancer external endpoint..."
                 sleep 10
```

Comment:
Looks to be that the script would sleep unnecessarily for at least 10 seconds in all cases.  The sleep is for retry purposes in the event that the status wasn't yet populated, which isn't relevant in the annotation case.   The fix would simply be to initialize the `PUBLISHED_PORT` where it is declared.

Reply:
Fixed